### DISCUSSION TITLE
Kill-switching in K8s

### DISCUSSION THREAD

**Discussion by DrewRubin** [qdm12/gluetun#1053] [Q&A]
Hello all,
In Docker, there is the concept of using `network_mode: "service:gluetun"` to force all traffic from another container to go through gluetun. In Kubernetes, we don't have that option to my knowledge, so we use sidecar containers. All containers in the pod that gluetun is in will use gluetun's networking in an ideal scenario, but what about in the case of a failure or non-ideal scenario? For example, there isn't much of a guarantee of container start order. There are certain workarounds to force it, but it's not really the "K8s way." So if my container that I want to route traffic only through the VPN starts first, couldn't that be a leak? Or perhaps let's say for one reason or another the gluetun container does not start and crashes repeatedly unbeknownst to me. What steps can be taken to mitigate these sorts of scenarios in K8s?

**Comment by qdm12** [comment]
I don't personally use kubernetes with Gluetun so I am not 100% knowledgeable on that;

Have you tried putting in bad settings in Gluetun to have in a crash loop, does the other container in the pod have internet access?

Alternatively maybe this would help: https://docs.k8s-at-home.com/guides/pod-gateway/
